src/update-manager.ts

```diff
@@ -123,15 +161,22 @@ export default class UpdateManager {
 
     var request
     var definitive = false
+    const meta = this.store.fetcher.appNode
+    // const kb = s
+
      // @ts-ignore passes a string to kb.each, which expects a term. Should this work?
-    var requests = kb.each(undefined, this.ns.link('requestedURI'), docpart(uri))
+    var requests = kb.each(undefined, this.ns.link('requestedURI'), docpart(uri), meta)
     var method: string
     for (var r = 0; r < requests.length; r++) {
       request = requests[r]
       if (request !== undefined) {
-        var response = kb.any(request, this.ns.link('response')) as Quad_Subject
-        if (request !== undefined) {
-          var wacAllow = kb.anyValue(response, this.ns.httph('wac-allow'))
+        const response = kb.any(request, this.ns.link('response'), null, meta) as Quad_Subject
+        if (response !== undefined) { // ts
+
+          const outOfDate = kb.anyJS(response, this.ns.link('outOfDate'), null, meta)
+          if (outOfDate) continue
+
+          var wacAllow = kb.anyValue(response, this.ns.httph('wac-allow'), null, meta)
```

Comment:
One thing I was having trouble grasping.. when it sees that is is outOfDate is there something it needs to do to "refresh" and become up to date? It's probably not the case but I feel like maybe we are assuming that if it's out of date that it is okay to continue.  I think I'm missing some key thing that happens when we login.

Reply:
Yes and no.  At that level, the editable function is synch so it can't replace the out of date data as it can't access the web.  But any ftehc which has been done since logic will noot be out of date.  The fomrs for example tend to fetch the data when they display a form.  But alos in future those who can use anyc functions can use checkEditable which first tries using existing data and if it find nothing which is not out of date, does a fetcher.load() of the thing, which will add new not out of data metadata, so the second call to editable() should work.